pkg/kb_agent/httpserver/server.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright (C) 2022-2024 ApeCloud Co., Ltd
+
+This file is part of KubeBlocks project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package httpserver
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	fasthttprouter "github.com/fasthttp/router"
+	"github.com/valyala/fasthttp"
+)
+
+// Server is an interface for the kb-agent HTTP server.
+type Server interface {
+	io.Closer
+	Router() fasthttp.RequestHandler
+	StartNonBlocking() error
+}
+
+type server struct {
+	config    Config
+	endpoints []Endpoint
+	servers   []*fasthttp.Server
+}
+
+// NewServer returns a new HTTP server.
+func NewServer() Server {
+	return &server{
+		endpoints: Endpoints(),
+		config:    config,
+	}
+}
+
+// StartNonBlocking starts a new server in a goroutine.
+func (s *server) StartNonBlocking() error {
+	logger.Info("Starting HTTP Server")
+	handler := s.Router()
+
+	APILogging := s.config.APILogging
+	if APILogging {
+		handler = s.apiLogger(handler)
+	}
+
+	var listeners []net.Listener
+	if s.config.UnixDomainSocket != "" {
```

Comment:
didn't see the code that initialize UnixDomainSocket, so it always equals to "".